Sources/Common/Communication/Event.swift

```diff
@@ -49,6 +49,29 @@ public enum EventTypesRegistry {
             NewSubscriptionEvent.self
         ]
     }
+
+    static func getEventType(for key: String) throws -> (any EventRepresentable.Type) {
+        switch key {
+        case ProfileIdentifiedEvent.key:
+            return ProfileIdentifiedEvent.self
+        case ScreenViewedEvent.key:
+            return ScreenViewedEvent.self
+        case ResetEvent.key:
+            return ResetEvent.self
+        case TrackMetricEvent.key:
+            return TrackMetricEvent.self
+        case TrackInAppMetricEvent.key:
+            return TrackInAppMetricEvent.self
+        case RegisterDeviceTokenEvent.key:
+            return RegisterDeviceTokenEvent.self
+        case DeleteDeviceTokenEvent.key:
+            return DeleteDeviceTokenEvent.self
+        case NewSubscriptionEvent.key:
+            return NewSubscriptionEvent.self
+        default:
+            throw EventBusError.invalidEventType
+        }
+    }
```

Comment:
It absolutely would be 😄, I think I started with it being an enum but pivoted because of the challenges it brought with inheritance, being codable and few other challenges